lib/auth.ts

```diff
@@ -175,17 +175,27 @@ const getAuth = function (
 		email: string;
 		password: string;
 	}): Promise<string> {
-		const { body } = await request.send<string>({
-			method: 'POST',
-			baseUrl: apiUrl,
-			url: '/login_',
-			body: {
-				username: credentials.email,
-				password: String(credentials.password),
-			},
-			sendToken: false,
-		});
-		return body;
+		try {
+			const { body } = await request.send<string>({
+				method: 'POST',
+				baseUrl: apiUrl,
+				url: '/login_',
+				body: {
+					username: credentials.email,
+					password: String(credentials.password),
+				},
+				sendToken: false,
+			});
+			return body;
+		} catch (err) {
+			if (err.statusCode === 401) {
+				throw new errors.BalenaInvalidLoginCredentials();
+			}
+			if (err.statusCode === 429) {
+				throw new errors.BalenaTooManyRequests();
+			}
+			throw new errors.BalenaRequestError('Unauthorized', err.statusCode, {});
```

Comment:
Imo we should re-throw the `err` since
* the server might have returned anything interesting, in which case we don't need to handle/conceal it
* what if `credentials` is undefined and it's just the js runtime that throws

PS: I think that when the `err` is from balena-request, it already a BalenaRequestError